subsys/bluetooth/controller/ll_sw/ull_connections.c

```diff
@@ -399,6 +399,49 @@ void ull_conn_done(struct node_rx_event_done *done)
 		return;
 	}
 
+#if defined(CONFIG_BT_CTLR_LE_ENC)
+	/* Check authenticated payload expiry or MIC failure */
+	switch (done->extra.mic_state) {
+	case LLL_CONN_MIC_NONE:
+#if defined(CONFIG_BT_CTLR_LE_PING)
+		if (lll->enc_rx || conn->llcp.enc.pause_rx) {
+			uint16_t appto_reload_new;
+
+			/* check for change in apto */
+			appto_reload_new = (conn->apto_reload >
+					    (lll->latency + 6)) ?
+					   (conn->apto_reload -
+					    (lll->latency + 6)) :
+					   conn->apto_reload;
+			if (conn->appto_reload != appto_reload_new) {
+				conn->appto_reload = appto_reload_new;
+				conn->apto_expire = 0U;
+			}
+
+			/* start authenticated payload (pre) timeout */
+			if (conn->apto_expire == 0U) {
+				conn->appto_expire = conn->appto_reload;
+				conn->apto_expire = conn->apto_reload;
+			}
+		}
+#endif /* CONFIG_BT_CTLR_LE_PING */
+		break;
+
+	case LLL_CONN_MIC_PASS:
+#if defined(CONFIG_BT_CTLR_LE_PING)
+		conn->appto_expire = conn->apto_expire = 0U;
+#endif /* CONFIG_BT_CTLR_LE_PING */
+		break;
+	/* TODO decide on how to handle this */
+	/*
+	case LLL_CONN_MIC_FAIL:
+		conn->llcp.terminate.reason_peer =
+			BT_HCI_ERR_TERM_DUE_TO_MIC_FAIL;
+		break;
+	*/
```

Comment:
`conn->llcp.terminate.reason_peer` has been renamed to `conn->terminate.reason`

Reply:
Renamed